### PR TITLE
Update `setup.py` and README for r2.1 release.

### PR DESCRIPTION
**Pull request by will-cromar**
- Backport build metadata and README changes from release branch
- Attempt to clean up various build lists
- Bury old versions in `<details>` to make the lists less overwhelming

### CODE REVIEW

File: README.md
Comment (JackCaoG) [inline README.md:271]: There is a place in README mentions `pip install torch_xla[tpuvm]`, can you add notes saying that only works for release prior to 2.0? for nightly do we use `[tpuvm]` or `[tpu]`?

Comment (will-cromar) [reply README.md:271]: I left `tpuvm` on master (see comment in `setup.py`) so I don't break people's scripts. We should remove it eventually, so I'd rather leave this as an undocumented "feature".

The instruction in this doc already says "before PyTorch/XLA 2.0"

**Review by JackCaoG — APPROVED** [review]

**Comment by will-cromar** [comment]
Since I touched `setup.py`, I'll wait until the build passes before I merge.

**Review by ManfeiBai — APPROVED** [review]
LGTM

File: README.md
Comment (vanbasten23) [inline README.md:150]: pytorch uses cuda 12.1 instead of 12.0: `pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/test/cu121`. Do we want to do the same?

File: setup.py
Comment (vanbasten23) [inline setup.py:359]: What's the difference between `pip install torch_xla[tpu] -f https://storage.googleapis.com/libtpu-releases/index.html` and `pip install torch_xla[tpuvm]`? IOW, under what circumstance should I use the former versus the latter?

File: README.md
Comment (will-cromar) [reply README.md:150]: Oh interesting. I just went off of what we had built. Do you know if our 12.0 wheel is compatible with their 12.1 wheel? Or do we need to add a new build config for the 2.0 release?

File: setup.py
Comment (will-cromar) [reply setup.py:359]: The technical difference is that `tpuvm` refers to a private package repository directly. That's why you don't have to pass an `--extra-index`. PyPI disallows this, so with release wheels starting with 2.1, you have to use `tpu`, which specifies a libtpu version but not the location. The location comes from the index url.

I left `tpuvm` on master/nightly for now since there's no rush to remove it, but I think we might want to anyway to reduce confusion.

If you're not sure, just use the `tpu` extra dependency since we will have to support that going forward.

Comment (vanbasten23) [reply setup.py:359]: Got it. Thanks!

File: README.md
Comment (vanbasten23) [reply README.md:150]: > Do you know if our 12.0 wheel is compatible with their 12.1 wheel?

If we do `pip3 install torch torchvision torchaudio
`, it'll install cuda 12.1 by default per https://screenshot.googleplex.com/BDFoq3C7TXiWeXn. I can do some testing.

But before that, we do need to add 2 new build config for the 2.1 release, one for cuda12.1+py3.8 and another one for cuda12.1+py3.10. Once we get the wheel and validate that it is compatible with pytorch's, we can update our GPU doc. Wdyt?

Comment (will-cromar) [reply README.md:150]: Sounds good to me. Thanks for catching the issue.

Comment (vanbasten23) [reply README.md:150]: hi @ManfeiBai , it seems we need cuda 12.1 wheels instead of 12.0 because by default PyTorch uses cuda 12.1. Can you please generate a py=3.8 cuda=12.1 wheel and a py=3.10 cuda=12.1 wheel for torch_xla release 2.1?

Comment (ManfeiBai) [reply README.md:150]: PR for CUDA12.1 wheel: https://github.com/pytorch/xla/pull/5683

**Pull request merged**
